modules/wallet/src/main/scala/org/tesselation/wallet/transaction/package.scala

```diff
@@ -0,0 +1,51 @@
+package org.tesselation.wallet
+
+import java.security.KeyPair
+
+import cats.effect.Async
+import cats.syntax.applicative._
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+
+import org.tesselation.crypto.Signed
+import org.tesselation.crypto.ops._
+import org.tesselation.ext.crypto._
+import org.tesselation.keytool.security.{SecurityProvider, WithSecureRandom}
+import org.tesselation.kryo.KryoSerializer
+import org.tesselation.schema.address.Address
+import org.tesselation.schema.transaction._
+
+import eu.timepit.refined.auto._
+import eu.timepit.refined.types.all.NonNegBigInt
+import io.estatico.newtype.ops._
+
+package object transaction {
+
+  implicit class TransactionOrdinalOps(ordinal: TransactionOrdinal) {
+
+    def next: TransactionOrdinal =
+      TransactionOrdinal(NonNegBigInt.unsafeFrom(ordinal.coerce + 1))
```

Comment:
have you tried `NonNegBitInt(ordinal.coerce + 1)` or `ordinal |+| 1`?

Reply:
There is no `Semigroup[NonNegBitInt]` in refined.cats :<

The first one will obviously throw at compile time because it is not literal (standard apply method works only for literals)

Reply:
Oh wait there is a `Monoid` ! :)

Reply:
If there is please remove the `semigroup` for `NonNegBigInt` from `OrphanInstances` trait.

Reply:
Ok there is not. It was for standard int